test/EndToEndSpec.hs

```diff
@@ -555,6 +729,53 @@ spec = around_ (inTempDirectoryNamed "foo") $ do
             other-modules:
                 Baz
           |]
+
+        it "accepts generated-other-modules and infers other-modules" $ do
+          touch "src/Main.hs"
+          touch "src/Foo.hs"
+          [i|
+          executables:
+            foo:
+              main: Main.hs
+              source-dirs: src
+              generated-other-modules: XYZ
+          |] `shouldRenderTo` Package "foo" "Simple" ">= 2.0" [i|
+            executable foo
+              main-is: Main.hs
+              hs-source-dirs:
+                  src
+              other-modules:
+                  Foo
+                  Paths_foo
+                  XYZ
+              autogen-modules:
+                  XYZ
+              default-language: Haskell2010
+          |]
+
+        it "accepts generated-other-modules with explicit other-modules" $ do
+          touch "src/Foo.hs"
+          touch "src/Bar.hs"
+          [i|
+          executables:
+            foo:
+              main: Main.hs
+              source-dirs: src
+              other-modules: Baz
+              generated-other-modules: XYZ
+          |] `shouldRenderTo` Package "foo" "Simple" ">= 2.0" [i|
+            executable foo
+              main-is: Main.hs
+              hs-source-dirs:
+                  src
+              other-modules:
+                  Baz
+                  XYZ
+              autogen-modules:
+                  XYZ
+              default-language: Haskell2010
+          |]
+
```

Comment:
Can you try to turn some of the acceptance tests into unit tests for e.g. `determineModules`.

The general direction I want to go with the tests is: Get rid of the integration tests in `ConfigSpec` and `RunSpec` and have acceptance tests in `EndToEndSpec` and unit tests instead.